src/fswAlgorithms/sensorInterfaces/scanningInstrumentController/scanningInstrumentController.h

```diff
@@ -0,0 +1,60 @@
+/*
+ ISC License
+
+ Copyright (c) 2023, Autonomous Vehicle Systems Lab, University of Colorado Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+
+#ifndef SCANNINGINSTRUMENTCONTROLLER_H
+#define SCANNINGINSTRUMENTCONTROLLER_H
+
+#include <stdint.h>
+#include "cMsgCInterface/AccessMsg_C.h"
+#include "cMsgCInterface/AttGuidMsg_C.h"
+#include "cMsgCInterface/DeviceStatusMsg_C.h"
+#include "cMsgCInterface/DeviceCmdMsg_C.h"
+#include "architecture/utilities/bskLogging.h"
+
+/*! @brief Module to perform continuous instrument control
+ */
+typedef struct {
+
+    double attErrTolerance; //!< Normalized MRP attitude error tolerance
+    unsigned int useRateTolerance; //!< Flag to enable rate error tolerance
+    double rateErrTolerance; //!< Rate error tolerance in rad/s
+    unsigned int controllerStatus;  //!< dictates whether or not the controller should be running
+
+    /* declare module IO interfaces */
+    AccessMsg_C accessInMsg;  //!< Ground location access
+    AttGuidMsg_C attGuidInMsg;  //!< Attitude guidance input message
+    DeviceStatusMsg_C deviceStatusInMsg;  //!< Device status input message
+    DeviceCmdMsg_C deviceCmdOutMsg;  //!< Device status output message
+
+    BSKLogger *bskLogger;  //!< BSK Logging
+}scanningInstrumentControllerConfig;
```

Comment:
:small_red_triangle_down: This `struct` name can be capitalized.

Reply:
@LorenzzoQM was having some trouble compiling the code if this is capitalized. It has to do with some SWIG magic where it checks this struct and matches it to the module name. If the names don't match, it throws an error. The module name is not capitalized, so I feel like this shouldn't be either. It's also consistent with how it's done across Basilisk (which may or may not mean we should keep it that way).

Reply:
Thanks for the reply and information @joaogvcarneiro . 👍 I'll poke around this area myself some more. For now, content 😄